src/modules/gateway/selectors.spec.ts

```diff
@@ -0,0 +1,283 @@
+import { Network } from '@dcl/schemas'
+import { NetworkGatewayType } from 'decentraland-ui/dist/components/BuyManaWithFiatModal/Network'
+import {
+  manaFiatGatewayPurchaseCompleted,
+  openManaFiatGatewayRequest
+} from './actions'
+import { MoonPayTransactionStatus } from './moonpay/types'
+import { INITIAL_STATE } from './reducer'
+import {
+  getState,
+  getError,
+  getLoading,
+  isFinishingPurchase,
+  isOpeningGateway,
+  getData,
+  getPendingManaPurchase,
+  getPurchases,
+  getNFTPurchase
+} from './selectors'
+import { TradeType } from './transak/types'
+import { Purchase, PurchaseStatus } from './types'
+
+let manaFiatGatewayState: any
+
+const mockManaPurchase: Purchase = {
+  address: '0x9c76ae45c36a4da3801a5ba387bbfa3c073ecae2',
+  amount: 100,
+  id: 'mock-id',
+  network: Network.ETHEREUM,
+  timestamp: 1535398843748,
+  status: PurchaseStatus.PENDING,
+  gateway: NetworkGatewayType.MOON_PAY,
+  txHash: null
+}
+
+const mockContractAddress = 'a-contract-address'
+const mockTokenId = 'aTokenId'
+
+const mockNFTPurchase: Purchase = {
+  ...mockManaPurchase,
+  amount: 1,
+  nft: {
+    contractAddress: mockContractAddress,
+    tokenId: mockTokenId,
+    tradeType: TradeType.PRIMARY,
+    cryptoAmount: 100
+  }
+}
+
+describe('MANA-FIAT Gateway selectors', () => {
+  beforeEach(() => {
+    manaFiatGatewayState = { gateway: INITIAL_STATE }
```

Comment:
This would be the application state, not just the manaFiatGatewayState.

Also, manaFiatGatewayState should be renamed to just gatewayState.